File: ruby/Problem005_SmallestMultiple.rb
require "prime"



array = [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]

(2..20).each do |i|
	i.prime_division.each do |j|
		if array[ j[0] ]  <  j[1]
			array[ j[0] ] = j[1]
		end
	end
end

puts array.inspect



out = 1

(2..20).each do |i|
	(0...array[i]).each do |j|
		out *= i
	end
end

puts out

   # 2520 is the smallest number that can be divided by each of the numbers from 1 to 10 without any remainder.
   # What is the smallest number that is evenly divisible by all of the numbers from 1 to 20?

File: java/Problem005_SmallestMultiple.java
public class Problem005_SmallestMultiple {
    public static void main (String args[]) { new Problem005_SmallestMultiple(); }

    public Problem005_SmallestMultiple() {
        int number = 1;
        for (int i=1; i<20; i++)
            number = lcm(number, i);

        System.out.println(number);
    }

    int gcd(int m, int n) {  // English  -  Greatest Common Divisor
        return (n == 0)? m : gcd(n, m % n);
    }   // Greatest  Common  Divisor
    int lcm(int m, int n) { return(m * n) / gcd(m, n); }            // Least     Common  Multiple
}


/*
   2520 is the smallest number that can be divided by each of the numbers from 1 to 10 without any remainder.

   What is the smallest number that is evenly divisible by all of the numbers from 1 to 20?
 */